sdk/src/test/java/com/finbourne/lusid/tutorials/ibor/Valuation.java

```diff
@@ -178,8 +262,136 @@ private void run_aggregation(
+        ListAggregationResponse aggregationResponse = aggregationApi.getAggregationByPortfolio(TutorialScope, portfolioId,null, null, null, aggregationRequest);
+
+        aggregationResponse.getData().sort((o1, o2) -> {
+            String name1 = (String)o1.get(groupingKey);
+            String name2 = (String)o2.get(groupingKey);
+
+            return name1.compareTo(name2);
+        });
+
+        List<Map<String, Object>> aggregationResults = aggregationResponse.getData();
+        logResults(groupingKey, aggregationResults);
+        return  aggregationResults;
+    }
+
+
+    /**
+     * Create a request to submit to LUSID to generate a transaction portfolio.
+     *
+     * @param portfolioId - unique portfolio id
+     * @param displayName - display name of the portfolio
+     * @param currency - base currency of the portfolio
+     * @return
+     * @throws ApiException
+     */
+    private CreateTransactionPortfolioRequest createTransactionPortfolioRequest(String portfolioId, String displayName, String currency) throws ApiException {
+        //  build the create portfolio request
+        CreateTransactionPortfolioRequest request = new CreateTransactionPortfolioRequest()
+                .displayName(displayName)
+                .code(portfolioId)
+                .baseCurrency(currency)
+                .created(EFFECTIVE_DATE);
+
+        return request;
+    }
+
+    /**
+     * Creates a market data entry quote request.
+     *
+     * @param instrument - market data instrument
+     * @param currency - base currency of the instrument
+     * @param value - value of the quote
+     * @return
+     */
+    private UpsertQuoteRequest createMarketDataUpsertQuoteRequest(String instrument, String currency, double value){
+        return new UpsertQuoteRequest()
+                .quoteId(new QuoteId()
+                        .quoteSeriesId(new QuoteSeriesId()
+                                .provider("DataScope")
+                                .instrumentId(instrument)
+                                .instrumentIdType(QuoteSeriesId.InstrumentIdTypeEnum.LUSIDINSTRUMENTID)
+                                .quoteType(QuoteSeriesId.QuoteTypeEnum.PRICE)
+                                .field("mid")
+                        )
+                        .effectiveAt(EFFECTIVE_DATE.toString())
+                )
+                .metricValue(new MetricValue().value(value).unit(currency));
+    }
+
+    /**
+     * Creates an fx entry quote request.
+     *
+     * @param baseCcy - base currency of the quote
+     * @param counterCcy - counter currency of the quote
+     * @param rate - the exchange rate
+     * @return
+     */
+    private UpsertQuoteRequest createFxUpsertQuoteRequest(String baseCcy, String counterCcy, double rate){
+        return new UpsertQuoteRequest()
+                .quoteId(new QuoteId()
+                        .quoteSeriesId(new QuoteSeriesId()
+                                .provider("DataScope")
```

Comment:
can you change this to `Lusid`, `DataScope` is a specific vendor provider

Reply:
Done. To do so I've also had to add LUSID as the FX market context supplier in the configuration recipe. Added relevant comment.